app/controllers/tokens_controller.rb

```diff
@@ -0,0 +1,17 @@
+class TokensController < ApplicationController
+  before_action :authenticate_user!
+
+  def generate_token
+    @user = current_user
+    token = SecureRandom.urlsafe_base64(25)
+    @user.update_columns(auth_token: token, token_created_at: Time.zone.now)
```

Comment:
why do you use #update_columns ?.
This skips all validations and callbacks

Reply:
changed to `assign_attributes` and used it in `before_create`